src/compiler/utilities.ts

```diff
@@ -4847,7 +4847,7 @@ namespace ts {
 
     function getDeclarationIdentifier(node: Declaration | Expression): Identifier | undefined {
         const name = getNameOfDeclaration(node);

```

Comment:
can we fix `getNameOfDeclaration` as well to return nullable type? i do not know where else our assumptions are wrong, and this seems like a good time as any to fix them once and for all.

Reply:
I'll try. It's used a lot of places without checking, but many of these follow a similar error-reporting pattern that might be supportable with a non-nullable overload or something.